nemo/collections/llm/modelopt/quantization/quantizer.py

```
@@ -354,13 +365,27 @@ def export(self, model, model_dir: str, trainer: Optional["Trainer"] = None) ->
                 TrainerContext.from_trainer(trainer).io_dump(ckpt_to_context_subdir(export_dir), yaml_attrs=["model"])
                 assert (Path(ckpt_to_weights_subdir(export_dir, False)) / "modelopt_state").exists()
         elif self.export_config.export_format == "hf":
-            assert is_automodel, "HF export is only supported for AutoModelForCausalLM"
             unwrapped_model = unwrap_for_modelopt_operations(model)
             with torch.inference_mode():
-                mte.export_hf_checkpoint(
-                    unwrapped_model,
-                    export_dir=export_dir,
-                )
+                if is_automodel:
+                    mte.export_hf_checkpoint(
+                        unwrapped_model,
+                        export_dir=export_dir,
+                    )
+                else:
+                    context = io.load_context(model_dir, subpath="model")
+                    try:
+                        exporter = context.exporter("hf", model_dir)
+                        config = exporter.config
+                        with tempfile.TemporaryDirectory() as tmp_dir:
+                            config.save_pretrained(tmp_dir)
+                            pretrained_path = Path(str(tmp_dir))
+                            mte.export_mcore_gpt_to_hf(unwrapped_model, pretrained_path, export_dir=str(export_dir))
+                    except Exception as e:
+                        # Mcore GPT Models are supported, but do not have HF config.
+                        if "No connector found for extension 'hf'" not in str(e):
```

Is there a more robust way of handling this case? Do we know what type of exception is raised here and use that instead of hard-coded string matching?

---

I've just checked that plain GPT model cannot be consumed by TRT-LLM / vLLM (no `config.json`). I removed this try-catch